modules/tinymce/src/core/test/ts/browser/EditorAutoFocusTest.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, it, before, afterEach } from '@ephox/bedrock-client';
+import { Insert, SugarBody, SugarElement } from '@ephox/sugar';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorManager } from 'tinymce/core/api/PublicApi';
+
+const isInViewport = (editor: Editor) => {
+  const containerRect = editor.getContentAreaContainer().getBoundingClientRect();
+  const innerRect = editor.selection.getNode().getBoundingClientRect();
+  const top = containerRect.top + innerRect.top + 1;
+  return top > 0 && top + innerRect.height < window.innerHeight;
+};
+
+const setupEditorAutoFocus = (id: string) => {
+  return new Promise((resolve) => {
+    EditorManager.init({
+      selector: 'textarea',
+      base_url: '/project/tinymce/js/tinymce/',
+      toolbar: '',
+      menubar: false,
+      statusbar: false,
+      height: 1000,
+      auto_focus: id,
+      init_instance_callback: (editor: Editor) => {
+        editor.on('focus', resolve);
+      }
+    });
+  });
+};
+
+const testEditorAutoFocus = async (id: string) => {
+  await setupEditorAutoFocus(id);
+  const editor = EditorManager.get(id);
+  assert.isTrue(editor.hasFocus());
+  assert.isTrue(isInViewport(editor));
+};
+
+describe('browser.tinymce.core.EditorAutoFocusTest', () => {
```

Comment:
We also need to make sure this works in inline mode correctly, so can you please add tests for that as well? I'd suggest using the `Arr.each` with `context` method I mentioned last night to do that.

- [ ] Add tests to make sure this works with inline mode as well